### ISSUE TITLE
 Goto definition goes to stub files instead of actual definitions

### ISSUE THREAD

**Issue by ewerybody** [microsoft/pylance-release#949]
Seems like the issue (#65) has not been solved entirely :/
Examples from the old report still show "some" preference to the stub files although there are source files in place.:
```python
from http import HTTPStatus

print(HTTPStatus.GATEWAY_TIMEOUT)
```
performing **Goto Definition**/**Goto Declaration** DOES work on `HTTPStatus` but always goes to Declaration on `http` and `GATEWAY_TIMEOUT`

```python
from operator import attrgetter
```
neither `operator` nor `attrgetter` bring you to the source code definition on **Goto Definition** or **Goto Declaration**

```python
from xml.etree import ElementTree
```
all goto the **Declaration stub** whatever I do

```python
import pipes
pipes.quote()
```
only `quote` works here.

## Expected
**IF** there is source code
**Goto Definition** brings you to the source code `.py`
**Goto Declaration** opens the stub `.pyi`

## Environment data
Version: 1.54.0-insider (user setup)
Commit: 93f705ab40b37aade9d3b5165ed09114a8c87ac9
Date: 2021-02-11T05:44:31.644Z (2 days ago)
Electron: 11.2.3
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042

**Comment by jakebailey** [comment]
#65 was about filtering the navigation, so that was closed. These above issues were with source mapping (mapping pyi contents to py content) and required a different fix.

All of the above will work in the next version, except `attrgetter`, because the type listed in the stub (function) doesn't match the type in the code (class) and our source mapper thinks it has found the wrong thing.

![gotothing3](https://user-images.githubusercontent.com/5341706/108559474-3a213f00-72b0-11eb-8b17-0076ac6be799.gif)

**Comment by heejaechang** [comment]
related to https://github.com/microsoft/pyrx/issues/1123

**Comment by jakebailey** [comment]
`attrgetter` now works, so everything you've listed will be fixed in our next release.

![gotothing4](https://user-images.githubusercontent.com/5341706/108568367-d3575200-72be-11eb-9973-18e4531557ec.gif)

**Comment by ewerybody** [comment]
whow! Thats great news!! 🤩

So the attrgetter stub with the `def` type is from the Python extension, right? Did they fix it as well?

**Comment by jakebailey** [comment]
That stub is sourced from typeshed and would need to be fixed there. I don't know what the history of that stub is to know if it was intentional or not. I've seen that sort of thing a few times.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.2.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202124-24-february-2021
